src/libYARP_sig/include/yarp/sig/PointCloud.h

```diff
@@ -0,0 +1,590 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_SIG_POINTCLOUD_H
+#define YARP_SIG_POINTCLOUD_H
+
+#include <yarp/sig/Vector.h>
+#include <yarp/sig/PointCloudNetworkHeader.h>
+
+#include <map>
+#include <vector>
+#include <cstring>
+
+
+namespace yarp {
+    namespace sig {
+        template <class T> class PointCloud;
+        class PointCloudBase;
+    }
+}
+
+namespace yarp {
+    namespace sig {
+    /**
+     * @brief The PointCloudBase class
+     *
+     * A Base class for a yarp::sig::PointCloud<T>, provide default implementation for
+     * some methods and hides some internal implementation to the user.
+     * It can't be instantiated, only inherited.
+     */
+    class YARP_sig_API PointCloudBase : public yarp::os::Portable
+    {
+    public:
```

Comment:
You should have a virtual destructor in this class, otherwise, if you want to delete an array of `PointCloudBase*` you will not call the right dtor.